### ISSUE TITLE
force reroute of (primary) shards from a specific node

### ISSUE THREAD

**Issue by yehosef** [elastic/elasticsearch#15681]
(perhaps this is possible - I couldn't figure out how.. - for reference, I'm using 1.7.4)

My situation is that I have a large cluster that I want to make smaller.  I would like to be able to route primary shards away from this node so when I take the node out, it minimizes the effect.  I realize I can do this manually with the reroute API but it doesn't support wildcards so I have do it all manually which is really tedious.   The levels of need/impact are:
1) primary shards with no replicas - I know it's not recommended, but we have some logging where the data is very big and we don't want to have to double the disk space.  If a machine goes down, we can reindex.  Let's say I have a 5 machine cluster and I want to take a node, I'd like to be able to move all the shards the other machines.  
2) primary shards with replicas - even though I won't lose any data (theoretically) there is still some "hiccuping" while the cluster determines that this primary shard is not available and a different shard is promoted and another replica is initialized.  Just turning off the node puts the cluster into a "failure mode" which is should be able to recover from.
3) replica shards, this causes less because you just need to create a new replica - I could live without this, but it would be nice.

The basic principle is that we shouldn't force the user to enter a failure mode with a planned node removal.  Failure modes in Elasticsearch can be problematic and there is no reason to go there needlessly.

I had though https://www.elastic.co/guide/en/elasticsearch/reference/current/shard-allocation-filtering.html might help - but it only seems to deal with allocation - not forcing reallocation.

**Comment by ywelsch** [comment]
It is all explained here:
https://www.elastic.co/guide/en/elasticsearch/reference/current/allocation-filtering.html

Please use discuss.elastic.co to ask these kinds of questions ;-)

**Comment by yehosef** [comment]
I did bring it up in discuss.elastic.co - https://discuss.elastic.co/t/force-shard-reallocation/36839, thank you very much ;-)

I tried the advice there and it did not _reallocate_ - but it did prevent allocation.  

But it's still not clear.  I read the docs more carefully (thanks) While the docs says:

>The typical use case for cluster-wide shard allocation filtering is when you want to decommision a node, and you would like to move the shards from that node to other nodes in the cluster before shutting it down.

It then goes on to say:

>Cluster-wide shard allocation filtering works in the same way as index-level shard allocation filtering (see Index Shard Allocation for details).

And the index level shard allocation did *not* reallocate shards.  

The docs for 1.7 at https://www.elastic.co/guide/en/elasticsearch/reference/1.7/index-modules-allocation.html seems to say this should work:

> The provided settings can also be updated in real time using the update settings API, allowing to "move" indices (shards) around in realtime.

But it didn't work for me.   I change the settings for a particular index to exclude on a node (tried both ip and host exclusion) and waited 15-20 minutes on an idle cluster but nothing happened.  

Any suggestions?
